firmware/src/sgx/src/untrusted/keyvalue_store.c

```diff
@@ -23,26 +23,54 @@
  */
 
 #include <sys/stat.h>
+#include <ctype.h>
 #include "hsm_u.h"
 #include "log.h"
 
 #define KVSTORE_PREFIX "./kvstore-"
 #define KVSTORE_SUFFIX ".dat"
+#define KVSTORE_MAX_KEY_LEN 256
+
+
+// Sanitizes a key by allowing only [a-zA-Z0-9]. Anything else is replaced by an underscore
+static void sanitize_key(char* key, char *sanitized_key) {
```

Comment:
I know we currently don't have unit tests for the key value store, but this sort of logic deserves a little bit of testing. What's the quickest way to achieve something like this?

Reply:
I agree, this deserves some test cases, just found it a bit of an overkill writing test for the entire module just for testing this. What do you think?

Maybe we can factor out just the file operations to their own module (`file_utils.c`) and write a simple test suite for this module, how does that sound?

Reply:
sounds good, thanks.